docs/core/extensions/sslstream-best-practices.md

```diff
@@ -0,0 +1,107 @@
+
+Consider an IoT device without a persistent clock. After powering on, the clock of the device would start many years in the past and, therefore, all certificates would be considered "not yet valid". The following code shows a validation callback implementation which ignores validity period violations.
+
+```csharp
+static bool CustomCertificateValidationCallback(
+    object sender,
+    X509Certificate? certificate,
+    X509Chain? chain,
+    SslPolicyErrors sslPolicyErrors)
+{
+    if (chain is null)
+    {
+        return false;
+    }
+
+    foreach (X509ChainStatus status in chain.ChainStatus)
+    {
+        if ((status.Status & ~X509ChainStatusFlags.NotTimeValid) != X509ChainStatusFlags.NoError)
+        {
+            return false;
+        }
+    }
+
+    return true;
+}
+```
+
+### Certificate pinning
+
+Another situation where custom certificate validation is necessary is when clients expect servers to use a specific certificate, or a certificate from a small set of known certificates. This practice is known as [certificate pinning](https://owasp.org/www-community/controls/Certificate_and_Public_Key_Pinning). The following code snippet shows a validation callback which checks that the server presents a certificate with a specific known public key.
+
+```csharp
+static bool CustomCertificateValidationCallback(
+    object sender,
+    X509Certificate? certificate,
+    X509Chain? chain,
+    SslPolicyErrors sslPolicyErrors)
+{
+    if (certificate == null)
+    {
+        return false;
+    }
+
+    const string ExpectedPublicKey =
+        "3082010A0282010100C204ECF88CEE04C2B3D850D57058CC9318EB5C" +
+        "A86849B022B5F9959EB12B2C763E6CC04B604C4CEAB2B4C00F80B6B0" +
+        "F972C98602F95C415D132B7F71C44BBCE9942E5037A6671C618CF641" +
+        "42C546D31687279F74EB0A9D11522621736C844C7955E4D16BE8063D" +
+        "481552ADB328DBAAFF6EFF60954A776B39F124D131B6DD4DC0C4FC53" +
+        "B96D42ADB57CFEAEF515D23348E72271C7C2147A6C28EA374ADFEA6C" +
+        "B572B47E5AA216DC69B15744DB0A12ABDEC30F47745C4122E19AF91B" +
+        "93E6AD2206292EB1BA491C0C279EA3FB8BF7407200AC9208D98C5784" +
+        "538105CBE6FE6B5498402785C710BB7370EF6918410745557CF9643F" +
+        "3D2CC3A97CEB931A4C86D1CA850203010001";
+
+    return certificate.GetPublicKeyString().Equals(ExpectedPublicKey);
+}
+```
+
+## Considerations for client certificate validation
+
+Server applications need to be careful when requiring and validating client certificates. Instead of sending the entire certificate chain, clients can configure the [AIA (Authority Information Access)](http://www.pkiglobe.org/auth_info_access.html) extension which specifies where the issuer certificate can be downloaded. The server will then download the issuer certificate when building the <xref:System.Security.Cryptography.X509Certificates.X509Chain> for the client certificate. Similarly, servers may need to contact external servers to ensure that the client certificate has not been revoked.
```

Comment:
The AIA is not specific to client certs IMHO. But I understand the example bellow is mainly for servers.